api/uniter/unit_test.go

```diff
@@ -532,6 +532,8 @@ func (s *unitSuite) TestConfigSettings(c *gc.C) {
 	err = s.apiUnit.SetCharmURL(s.wordpressCharm.URL())
 	c.Assert(err, jc.ErrorIsNil)
 
+	s.WaitForModelWatchersIdle(c, s.State.ModelUUID())
```

Comment:
I think we need to verify that the test works _without_ this sync.

Reply:
The WaitForModelWatchersIdle does all the things that the StartSync would do. I have run this under stress for a while to confirm.

Reply:
But `SetCharmURL` should be synchronous with respect to cache population. Inserting the wait doesn't flex that assertion, because if it were not working as designed, the short pause might still cause the test to pass, to-wit the test should be more brutal.